src/perlin.js

```diff
@@ -381,5 +381,48 @@
     var nz = module.fbm3(x + 7.1, y + 4.9, z + 1.3, o);
     return module.fbm3(x + wa * nx, y + wa * ny, z + wa * nz, o, f, l);
   };
+
+  function hash(x, y) {
+    x = x & 255;
+    y = y & 255;
+    //return new Grad(Math.abs(Math.sin(x*123.657+y*46.87)*423548.15763) % 1, Math.abs(Math.sin(x*87.391+y*157.597)*587654.26461)%1, 0);
+    return new Grad(perm[x + perm[y]] / 255.0, perm[y + perm[x]] / 255.0, 0);
+  }
```

Comment:
That's a new function to this PR, previously it didn't exist! If you mean the commented out function I accidentally left that in there, I played around with using a procedural hash but it is way slower. Plus we have a pre-computed table of random numbers to use. The hash function I use there is actually standard for the type of use. It comes from Perlin's original noise implementation